### PR TITLE
Frontend: com_users view review

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Frontend: com_users view review

### Testing Instructions

Install testing sample data and test the com_users views

### Expected result

works

### Actual result

works with cs errors

### Documentation Changes Required

none

### CODE REVIEW

File: components/com_users/views/login/tmpl/default_login.php [hunk 1]
```diff
@@ -89,19 +80,21 @@
...
-			<?php echo JText::_('COM_USERS_LOGIN_REMIND'); ?></a>
-		</li>
-		<?php
-		$usersConfig = JComponentHelper::getParams('com_users');
-		if ($usersConfig->get('allowUserRegistration')) : ?>
-		<li>
-			<a href="<?php echo JRoute::_('index.php?option=com_users&view=registration'); ?>">
-				<?php echo JText::_('COM_USERS_LOGIN_REGISTER'); ?></a>
+				<?php echo JText::_('COM_USERS_LOGIN_REMIND'); ?>
+			</a>
```
Comment (brianteeman) [inline components/com_users/views/login/tmpl/default_login.php:95]: you have two opening a tags and only one closing

Comment (brianteeman) [reply components/com_users/views/login/tmpl/default_login.php:95]: sorry -misread it - its fine

File: components/com_users/views/profile/tmpl/default_params.php [hunk 2]
```diff
@@ -12,27 +12,30 @@
...
-			<?php endif; ?>
-		</dd>
-		<?php endif; ?>
-	<?php endforeach; ?>
-	</dl>
-</fieldset>
+	<fieldset id="users-profile-custom">
+		<legend><?php echo JText::_('COM_USERS_SETTINGS_FIELDSET_LABEL'); ?></legend>
+		<dl class="dl-horizontal">
+			<?php foreach ($fields as $field) :
```
Comment (C-Lodder) [inline components/com_users/views/profile/tmpl/default_params.php]: Shouldn't this:

```
<?php foreach ($fields as $field) :
	if (!$field->hidden) : ?>
```

be this:

```
<?php foreach ($fields as $field) : ?>
	<?php if (!$field->hidden) : ?>
```

?

Comment (zero-24) [reply components/com_users/views/profile/tmpl/default_params.php]: Thanks fixed

File: components/com_users/views/login/tmpl/default_login.php [hunk 3]
```diff
@@ -63,22 +57,19 @@
...
 				</div>
 			<?php endif; ?>
-
 			<?php if (JPluginHelper::isEnabled('system', 'remember')) : ?>
-			<div  class="control-group">
-				<div class="control-label"><label><?php echo JText::_('COM_USERS_LOGIN_REMEMBER_ME') ?></label></div>
-				<div class="controls"><input id="remember" type="checkbox" name="remember" class="inputbox" value="yes"/></div>
-			</div>
+				<div  class="control-group">
+					<div class="control-label"><label><?php echo JText::_('COM_USERS_LOGIN_REMEMBER_ME') ?></label></div>
```
Comment (Quy) [inline components/com_users/views/login/tmpl/default_login.php]: Add semicolon after `JText::_('COM_USERS_LOGIN_REMEMBER_ME')`

File: components/com_users/views/login/tmpl/default_login.php [hunk 4]
```diff
@@ -63,22 +57,19 @@
 					</div>
 				</div>
 			<?php endif; ?>
-
 			<?php if (JPluginHelper::isEnabled('system', 'remember')) : ?>
-			<div  class="control-group">
-				<div class="control-label"><label><?php echo JText::_('COM_USERS_LOGIN_REMEMBER_ME') ?></label></div>
-				<div class="controls"><input id="remember" type="checkbox" name="remember" class="inputbox" value="yes"/></div>
-			</div>
+				<div  class="control-group">
```
Comment (Quy) [inline components/com_users/views/login/tmpl/default_login.php]: Remove extra space.

File: components/com_users/views/login/tmpl/default_login.php [hunk 5]
```diff
@@ -12,33 +12,28 @@
...
-
 	<?php if (($this->params->get('logindescription_show') == 1 && str_replace(' ', '', $this->params->get('login_description')) != '') || $this->params->get('login_image') != '') : ?>
-	<div class="login-description">
+		<div class="login-description">
+	<?php endif; ?>
+	<?php if ($this->params->get('logindescription_show') == 1) : ?>
+		<?php echo $this->params->get('login_description'); ?>
+	<?php endif; ?>
+	<?php if ($this->params->get('login_image') != '') : ?>
+		<img src="<?php echo $this->escape($this->params->get('login_image')); ?>" class="login-image" alt="<?php echo JText::_('COM_USERS_LOGIN_IMAGE_ALT'); ?>"/>
```
Comment (n9iels) [inline components/com_users/views/login/tmpl/default_login.php]: Space missing before `/>`

File: components/com_users/views/login/tmpl/default_login.php [hunk 6]
```diff
@@ -63,22 +57,19 @@
...
 			<?php endif; ?>
-
 			<?php if (JPluginHelper::isEnabled('system', 'remember')) : ?>
-			<div  class="control-group">
-				<div class="control-label"><label><?php echo JText::_('COM_USERS_LOGIN_REMEMBER_ME') ?></label></div>
-				<div class="controls"><input id="remember" type="checkbox" name="remember" class="inputbox" value="yes"/></div>
-			</div>
+				<div class="control-group">
+					<div class="control-label"><label><?php echo JText::_('COM_USERS_LOGIN_REMEMBER_ME'); ?></label></div>
+					<div class="controls"><input id="remember" type="checkbox" name="remember" class="inputbox" value="yes"/></div>
```
Comment (n9iels) [inline components/com_users/views/login/tmpl/default_login.php]: Space before closing tag of the input.

Or maybe change it like this
```php
<div class="controls">
	<input id="remember" type="checkbox" name="remember" class="inputbox" value="yes" />
</div>
```

File: components/com_users/views/profile/tmpl/edit.php [hunk 7]
```diff
@@ -42,99 +42,100 @@
...
+					<div class="clearfix"></div>
+				<?php endif; ?>
+			</fieldset>
+		<?php endif; ?>
 		<div class="control-group">
 			<div class="controls">
-				<button type="submit" class="btn btn-primary validate"><span><?php echo JText::_('JSUBMIT'); ?></span></button>
-				<a class="btn" href="<?php echo JRoute::_('index.php?option=com_users&view=profile'); ?>" title="<?php echo JText::_('JCANCEL'); ?>"><?php echo JText::_('JCANCEL'); ?></a>
+				<button type="submit" class="btn btn-primary validate">
+					<span>
```
Comment (Quy) [inline components/com_users/views/profile/tmpl/edit.php]: Remove `<span>` and `</span>` to be consistent with https://github.com/joomla/joomla-cms/pull/17236/files#diff-ce79943a89396b452660625e32cc7de8R44

**Comment by zero-24** [comment]
Fixed thanks!

File: components/com_users/views/profile/tmpl/edit.php [hunk 8]
```diff
@@ -42,99 +42,98 @@
...
+							<div class="control-group">
+								<div class="control-label">
+									<?php echo $field->label; ?>
+									<?php if (!$field->required && $field->type !== 'Spacer') : ?>
+										<span class="optional"><?php echo JText::_('COM_USERS_OPTIONAL'); ?></span>
+									<?php endif; ?>
+								</div>
+								<div class="controls">
+									<?php if ($field->fieldname === 'password1') : ?>
+										<?php // Disables autocomplete ?> <input type="password" style="display:none">
```
Comment (Quy) [inline components/com_users/views/profile/tmpl/edit.php]: Move comment to its own line.

File: components/com_users/views/login/tmpl/default_login.php [hunk 9]
```diff
@@ -63,22 +57,21 @@
...
 				</div>
 			<?php endif; ?>
-
 			<?php if (JPluginHelper::isEnabled('system', 'remember')) : ?>
-			<div  class="control-group">
-				<div class="control-label"><label><?php echo JText::_('COM_USERS_LOGIN_REMEMBER_ME') ?></label></div>
-				<div class="controls"><input id="remember" type="checkbox" name="remember" class="inputbox" value="yes"/></div>
-			</div>
+				<div class="control-group">
+					<div class="control-label"><label><?php echo JText::_('COM_USERS_LOGIN_REMEMBER_ME'); ?></label></div>
```
Comment (Quy) [inline components/com_users/views/login/tmpl/default_login.php]: Reformat.

**Comment by zero-24** [comment]
thanks @Quy fixed

File: components/com_users/views/login/tmpl/default_logout.php [hunk 10]
```diff
@@ -8,36 +8,35 @@
...
-
 	<?php if (($this->params->get('logoutdescription_show') == 1 && str_replace(' ', '', $this->params->get('logout_description')) != '')|| $this->params->get('logout_image') != '') : ?>
-	<div class="logout-description">
+		<div class="logout-description">
+	<?php endif; ?>
+	<?php if ($this->params->get('logoutdescription_show') == 1) : ?>
+		<?php echo $this->params->get('logout_description'); ?>
+	<?php endif; ?>
+	<?php if ($this->params->get('logout_image') != '') : ?>
+		<img src="<?php echo $this->escape($this->params->get('logout_image')); ?>" class="thumbnail pull-right logout-image" alt="<?php echo JText::_('COM_USER_LOGOUT_IMAGE_ALT'); ?>"/>
```
Comment (Quy) [inline components/com_users/views/login/tmpl/default_logout.php]: Add space before `/>`

File: components/com_users/views/profile/tmpl/edit.php [hunk 11]
```diff
@@ -42,99 +42,99 @@
...
+			<?php if (count($fields)) : ?>
+				<fieldset>
+					<?php // If the fieldset has a label set, display it as the legend. ?>
+					<?php if (isset($fieldset->label)) : ?>
+						<legend>
+							<?php echo JText::_($fieldset->label); ?>
+						</legend>
+					<?php endif; ?>
+					<?php if (isset($fieldset->description) && trim($fieldset->description)) : ?>
+						<?php echo '<p>' . $this->escape(JText::_($fieldset->description)) . '</p>'; ?>
```
Comment (Quy) [inline components/com_users/views/profile/tmpl/edit.php]: Change to `<p><?php echo $this->escape(JText::_($fieldset->description)); ?></p>`

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 6bedc973d3edbc73c9ac7ce5b2e2a85213052280<br /><br />Code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17236">issues.joomla.org/tracker/joomla-cms/17236</a>.</sub>

**Comment by n9iels** [comment]
I have tested this item :white_check_mark: successfully on 4de81fef1f2fcfe7b0ebdb023a5425f15b22e4e2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17236">issues.joomla.org/tracker/joomla-cms/17236</a>.</sub>

**Comment by ghost** [comment]
@Quy can you please retest?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 4de81fef1f2fcfe7b0ebdb023a5425f15b22e4e2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17236">issues.joomla.org/tracker/joomla-cms/17236</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Pull request merged**